File: backend/models/table.go
package models

import "time"

// Table テーブル（座席）情報
type Table struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	TableNumber int       `gorm:"uniqueIndex;not null" json:"table_number"` // 1, 2, 3...
	Capacity    int       `json:"capacity"`                                  // 座席数
	Status      string    `json:"status"`                                    // "active" or "inactive"
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: frontend/src/types/table.ts
// テーブル（座席）の型定義
export interface Table {
  id: number
  table_number: number
  capacity: number
  status: 'active' | 'inactive'
  created_at: string
  updated_at: string
}

// テーブル作成リクエスト
export interface CreateTableRequest {
  table_number: number
  capacity: number
  status?: 'active' | 'inactive'
}

// テーブル更新リクエスト
export interface UpdateTableRequest {
  table_number?: number
  capacity?: number
  status?: 'active' | 'inactive'
}

// 注文型にテーブル情報を追加
export interface Order {
  id: number
  user_id?: number
  product_id: number
  quantity: number
  total_price: number
  status: string
  table_id?: number
  created_at: string
  updated_at: string
  table?: Table
  product?: {
    id: number
    name: string
    price: number
    image?: string
  }
  user?: {
    id: number
    username: string
  }
}
